bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/config2/config.scala

```diff
@@ -0,0 +1,70 @@
+package edu.gemini.spModel.config2
+
+import scala.reflect.ClassTag
+
+import scalaz._
+import Scalaz._
+
+/** A bit of syntax to simplify working with Config.
+  */
+trait ConfigSyntax {
+  implicit final def configSyntax(c: Config): ConfigOps =
+    new ConfigOps(c)
+}
+
+object ConfigSyntax extends ConfigSyntax
+
+final class ConfigOps(val c: Config) extends AnyVal {
+  private def unsafe[A](e: String \/ A): A =
+    e.fold(s => sys.error(s), identity)
+
+
+  def extractOption[A](k: ItemKey)(implicit ev: ClassTag[A]): String \/ Option[A] =
+    Option(c.getItemValue(k)).fold(Option.empty[A].right[String]) { a =>
+      \/.fromTryCatchNonFatal {
+        Some(ev.runtimeClass.cast(a).asInstanceOf[A])
+      }.leftMap(_ => s"Expected '$k' to have type ${ev.runtimeClass.getName}")
+    }
```

Comment:
There's a similar `ConfigExtractor` in the ITC code, but we really need this type of thing in multiple places so having a version in `pot` makes sense I believe.  Ideally `ConfigExtractor` would be updated to delegate to this version or something.  What is the seqexec doing to get typed values out of the sequence?

There's probably a correct way to do the casting, and I'd be curious to know what it is.  After searching a bit I couldn't come up with anything better than what Florian did in `ConfigExtractor`.

Reply:
i think we took Florian's version and copied it but it has evolved ever since. We could use your version to reduce duplication